Firebase/DynamicLinks/FDLURLComponents/FDLURLComponents.m

```diff
@@ -448,15 +449,50 @@ - (instancetype)init {
 
 @implementation FIRDynamicLinkComponents
 
+#pragma mark Deprecated Initializers.
 + (instancetype)componentsWithLink:(NSURL *)link domain:(NSString *)domain {
-  return [[self alloc] initWithLink:link domain:domain];
+  NSString *domainURIPrefix = [NSString stringWithFormat:@"https://%@", domain];
```

Comment:
Is there any chance that a developer could already pass in a domain with `https://` in front? What happens if the site doesn't support `https`?

Reply:
We mandate https as the only acceptable scheme (as decided in api review doc). The previous methods always had the domain name only  (without scheme), but valid point that developers might sometimes get confused now. Added extra check.

Reply:
Done.